### PR TITLE
updates and saucelabs desktop support

### PR DESCRIPTION
**Pull request by patrickkettner**

### CODE REVIEW

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 40f9089c069e74ed934f61f7f155426d66213382
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58881539

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 40f9089c069e74ed934f61f7f155426d66213382
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58881539

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 40f9089c069e74ed934f61f7f155426d66213382
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58881539

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 3244279b9bff77f8d9a3b48ac6c4a5831eced82b
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58882605

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 3244279b9bff77f8d9a3b48ac6c4a5831eced82b
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58882605

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 48f804aac46fb13eefc7b6a19869552361e9aeee
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58884361

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: af9a6a2a99b1ef62b4c4535fa0ad7896b20e3a3c
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58884773

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 11dc14530057f096dc960f6939358a057f38723e
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58885191

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0acafc43181658b062e59557c85b34e67938e32e
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58885583

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by modernizr-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 559c4c792bc602feec4c0acc1a4ab13334d679f9
Build details: https://travis-ci.org/modernizr-savage/Modernizr/builds/58999873

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by patrickkettner** [comment]
@ryanseddon @SlexAxton @stucox @paulirish 



this commit ended up being a lot larger than I imagined for a few reasons.

1. there were ambient failures that kept cropping up - using new contexts for all tests seem to have finished all of those.
2. opera 12 hangs on the `mochaRunner` object that is used for the sauce test report. As a result, I had to swap out for the generic runner
3. due to a crapload of sauce running, I added `grunt-if` to only run saucelabs builds for a single node version, on a single OS.
4. introduced [savage](https://github.com/twbs/savage), which lets us run saucelabs for PRs. (you probably noticed the `modernizr-savage` user). Its currently running on my personal server, but I figure it can be moved to the new .com server once goes live soon.

could ya'll give this a once over and tell me if anything looks bad?

**Comment by patrickkettner** [comment]
also....

coveralls.io badge at 100%

**Comment by ryanseddon** [comment]
Looking now...

File: feature-detects/css/filters.js [hunk 1]
```diff
@@ -16,16 +16,26 @@ define(['Modernizr', 'createElement', 'prefixes', 'test/css/supports'], function
...
-        window.CSS.supports('filter', 'url()') :
-        window.supportsCSS('filter', 'url()');
+      var len = prefixes.length;
+      var prop = 'filter';
+      var supports;
+
+      for (var i = 0; i < len; i++) {
+        var prefixProp = prefixes[i] + prop;
+        supports = 'CSS' in window ?
+          window.CSS.supports(prefixProp, 'blur(2px)') :
```
Comment (ryanseddon) [inline feature-detects/css/filters.js]: Could we pull in the `nativeTestProps` here to check that?

File: test/browser/integration/caniuse.js [hunk 2]
```diff
@@ -150,29 +150,59 @@ window.caniusecb = function(caniuse) {
...
 
       // change the *documented* false positives
-      if (!ciubool && (o.feature == 'textshadow' && o.browser == 'firefox' && o.version == 3)) {
+      if (!ciubool && (o.feature == 'textshadow' && o.browser == 'Firefox' && o.version == 3)) {
         ciubool = o.fp = true;
       }
 
+      // firefox only supports web animation when a flag is enabled, which we
+      // don't do on sauce
+      if (o.feature === 'webanimations' && o.caniuseResult.indexOf('a d') === 0) {
```
Comment (ryanseddon) [inline test/browser/integration/caniuse.js]: Just so I understand, whats the indexOf check doing for these single characters?

File: test/browser/sauce-browsers.json [hunk 3]
```diff
@@ -1,38 +1,90 @@
...
   },
+
+
   {
-    "browserName": "safari",
-    "platform": "OS X 10.6",
-    "version": "5"
+    "browserName": "chrome",
+    "platform": "WIN8",
+    "version": "39"
```
Comment (ryanseddon) [inline test/browser/sauce-browsers.json]: Shame there is no way to say latest stable + the previous versions. Guess we'll just need to keep on top of this.

**Comment by ryanseddon** [comment]
LGTM other than the few comments left.

File: feature-detects/css/filters.js (hunk 1 above)
Comment (patrickkettner) [inline feature-detects/css/filters.js]: hah. yeah. great catch

File: test/browser/integration/caniuse.js (hunk 2 above)
Comment (patrickkettner) [inline test/browser/integration/caniuse.js]: its mappings to caniuse's dataset

```
y - (Y)es, supported by default
a - (A)lmost supported (aka Partial support)
n - (N)o support, or disabled by default
p - No support, but has (P)olyfill
u - Support (u)nknown
x - Requires prefi(x) to work
d - (D)isabled by default (need to enable flag or something)
#n - Where n is a number, starting with 1, corresponds to the notes_by_num note. For example: "42":"y #1" means version 42 is supported by default and see note 1.
```

File: test/browser/sauce-browsers.json (hunk 3 above)
Comment (patrickkettner) [inline test/browser/sauce-browsers.json]: I am gonna make a node module soon that takes [browserlist](https://www.npmjs.com/package/browserslist) and combines it with saucelabs browser's endpoints. that way we can say something like `last 3 versions`, etc, and avoid this

File: test/browser/integration/caniuse.js (hunk 2 above)
Comment (ryanseddon) [inline test/browser/integration/caniuse.js]: Ah thanks and added a comment block, good one!

File: test/browser/sauce-browsers.json (hunk 3 above)
Comment (ryanseddon) [inline test/browser/sauce-browsers.json]: That is a brilliant idea!

**Pull request merged**